hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx

```diff
@@ -608,90 +670,98 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
         """
-        cdef:
-            bint is_buy = active_order.is_buy
-            object order_price = active_order.price
-            object order_quantity = active_order.quantity
-            object suggested_price = self.c_get_market_making_price(market_pair, is_buy, order_quantity)
+        is_buy = active_order.is_buy
+        order_price = active_order.price
+        order_quantity = active_order.quantity
+        suggested_price = await self.get_market_making_price(market_pair, is_buy, order_quantity)
 
         if suggested_price != order_price:
 
             if is_buy:
-                if self._logging_options & self.OPTION_LOG_ADJUST_ORDER:
+                if self.OPTION_LOG_ADJUST_ORDER in self._logging_options:
                     self.log_with_clock(
                         logging.INFO,
                         f"({market_pair.maker.trading_pair}) The current limit bid order for "
                         f"{active_order.quantity} {market_pair.maker.base_asset} at "
                         f"{order_price:.8g} {market_pair.maker.quote_asset} is now below the suggested order "
                         f"price at {suggested_price}. Going to cancel the old order and create a new one..."
                     )
-                self.c_cancel_order(market_pair, active_order.client_order_id)
+                self.cancel_order(market_pair, active_order.client_order_id)
                 self.log_with_clock(logging.DEBUG,
                                     f"Current buy order price={order_price}, "
                                     f"suggested order price={suggested_price}")
                 return False
             else:
-                if self._logging_options & self.OPTION_LOG_ADJUST_ORDER:
+                if self.OPTION_LOG_ADJUST_ORDER in self._logging_options:
                     self.log_with_clock(
                         logging.INFO,
                         f"({market_pair.maker.trading_pair}) The current limit ask order for "
                         f"{active_order.quantity} {market_pair.maker.base_asset} at "
                         f"{order_price:.8g} {market_pair.maker.quote_asset} is now below the suggested order "
                         f"price at {suggested_price}. Going to cancel the old order and create a new one..."
                     )
-                self.c_cancel_order(market_pair, active_order.client_order_id)
+                self.cancel_order(market_pair, active_order.client_order_id)
                 self.log_with_clock(logging.DEBUG,
                                     f"Current sell order price={order_price}, "
                                     f"suggested order price={suggested_price}")
                 return False
 
         return True
 
-    cdef c_check_and_hedge_orders(self, object market_pair):
+    async def check_and_hedge_orders(self, market_pair: MakerTakerMarketPair):
```

Comment:
If the only difference between the processing we do for a gateway connector and a centralized connector is how we get the price, that should be extracted to a component to make this logic more generic.
Also it seems possible to unify the logic applied for buy and sells, since it differs in a couple of elements. A refactoring could be applied to parameterized those elements based on the order type, and provide them to the unified processing logic.

Reply:
Should changing how the gateway connector is implemented be a part of this PR?
Otherwise I've condensed the buys and sells.